plugins/modules/aci_vmm_vswitch.py

```diff
@@ -0,0 +1,412 @@
+  returned: info
+  type: list
+  sample:
+    {
+        "fvTenant": {
+            "attributes": {
+                "descr": "Production environment"
+            }
+        }
+    }
+previous:
+  description: The original configuration from the APIC before the module has started
+  returned: info
+  type: list
+  sample:
+    [
+        {
+            "fvTenant": {
+                "attributes": {
+                    "descr": "Production",
+                    "dn": "uni/tn-production",
+                    "name": "production",
+                    "nameAlias": "",
+                    "ownerKey": "",
+                    "ownerTag": ""
+                }
+            }
+        }
+    ]
+proposed:
+  description: The assembled configuration from the user-provided parameters
+  returned: info
+  type: dict
+  sample:
+    {
+        "fvTenant": {
+            "attributes": {
+                "descr": "Production environment",
+                "name": "production"
+            }
+        }
+    }
+filter_string:
+  description: The filter string used for the request
+  returned: failure or debug
+  type: str
+  sample: ?rsp-prop-include=config-only
+method:
+  description: The HTTP method used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec
+
+# via UI vSwitch Policy can only be added for VMware and Microsoft vmm domains
+# behavior for other domains is currently untested.
+VM_PROVIDER_MAPPING = dict(
+    cloudfoundry='CloudFoundry',
+    kubernetes='Kubernetes',
+    microsoft='Microsoft',
+    openshift='OpenShift',
+    openstack='OpenStack',
+    redhat='Redhat',
+    vmware='VMware',
+)
+
+enhanced_lag_spec = dict(
+    name=dict(type='str', required=True),
+    lacp_mode=dict(type='str', choices=['active', 'passive']),
+    load_balancing_mode=dict(
+        type='str',
+        choices=['dst-ip', 'dst-ip-l4port', 'dst-ip-vlan', 'dst-ip-l4port-vlan', 'dst-mac', 'dst-l4port',
+                 'src-ip', 'src-ip-l4port', 'src-ip-vlan', 'src-ip-l4port-vlan', 'src-mac', 'src-l4port',
+                 'src-dst-ip', 'src-dst-ip-l4port', 'src-dst-ip-vlan', 'src-dst-ip-l4port-vlan', 'src-dst-mac',
+                 'src-dst-l4port', 'src-port-id', 'vlan']),
+    number_uplinks=dict(type='int'),
+)
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(
+        port_channel_policy=dict(type='str'),
```

Comment:
The relationships like mtu policy used here is vmware domain specific, since  Microsoft is mentioned  adding  relationships specific to microsoft domain will be relevant

Reply:
the only missing policy for microsoft would be `vmmRsVswitchOverrideStpPo`l  .I can add that if you want. Same for  `vmmRsVswitchExporterPol`  (Netflow exporter policy for VMware) which I decided to omit because of rare usage based on my experience.

General question, what is the guidance on which config properties / relationships to expose ? I see that  e.g. in  the terraform usually all config properties are exposed, whereas in many of the ansible modules some config params are ommitted from the user-interface. 
Is there a consensus which parameters would be usually ommitted?

Reply:
Which relationships should be exposed depends on properties available on UI.